### DISCUSSION TITLE
SenderId mismatch | Cloud messaging Firebase

### DISCUSSION THREAD

**Discussion by deepesharora** [kreait/firebase-php#863] [Q&A]
hi Jerome, thanks for building this awesome library. I'm trying to integrate it into a test react-native project. 

### background & problem
I'm using it for Cloud messaging and it seems i've got my code to work with Android devices however when trying to send messages to FCM_tokens of ios devices I get the error below. 
```
Kreait\Firebase\Exception\Messaging\AuthenticationError
SenderId mismatch
``` 
I will appreciate you taking a quick look and pointing me in the right direction. I've been battling with this for 2 days. I do suspect its something on my laravel server side but unsure where to look. 

### My simple test code looks like this ....

```
##$message = CloudMessage::withTarget('token', 'ios_device_token_here') //<< when i use this i get the error. All i'm doing is change the fcm_token of the device, no other change in the code.

$message = CloudMessage::withTarget('token', 'android_device_token_here') // << when I use this the message goes to the phone/device

    ->withNotification($notification)
     /// i realize this APNs section is only relevant for ios but it seems to work even if left around for android, so left it here for the purposes of my test here. I intend to make this conditional later once i have the issue sorted.
    ->withApnsConfig(ApnsConfig::fromArray([

        'headers' => [

            'apns-priority' => '10',

        ],

        'payload' => [

            'aps' => [

                'badge' => 1,

                'sound' => 'default',

            ],

        ],

    ]));

 

  return $messaging->send($message);


```

**Comment by jeromegamez** [comment]
Do you have multiple Firebase projects, one for testing and one in production? I'm asking because the first thought popping into my mind was that the target FCM Tokens might relate to another project than the one you're sending the messages from.

Unfortunately I can only guess - the only Firebase Project I have is the one I use for the SDK tests, and there I can only work with tokens from web apps (I don't have an Android or iOS app to test with).

I believe connecting the APNS works by downloading a key from Apple and uploading it into your Firebase project (or something similar to that), right?

Can this key expire? Even if not, could you try generating a new key and thus re-connecting Firebase and the APNS?

And just to be sure: are you using the latest versions of the SDK/the Laravel package?

Sending to Android devices worked, did I understand this correctly?

**Reply by deepesharora** [reply]
Thanks for your response! 
Replies to your questions...

No, I have just 1 project.

You are correct. The APNS works by downloading a key from Apple and uploading it into your Firebase project.  No this key does not expire. I will try to reconnect Firebase to APN with a new key nonetheless.

I am on Laravel 6. and the compatible version of your package is 
```
##this is an excerpt from my composer.lock file
            "name": "kreait/firebase-php",
            "version": "5.26.5",
            ...
            ...
            "name": "kreait/laravel-firebase",
            "version": "3.4.0",
            ...
            ...
            "name": "laravel/framework",
            "version": "v6.20.44",
            
```

Yes, sending to Android devices worked!
